pyomo/contrib/pyros/pyros.py

```diff
@@ -79,13 +80,13 @@ class SolverResolvable(object):
     def __call__(self, obj):
         '''
         if obj is a string, return the Solver object for that solver name
-        if obj is a Solver object, return the Solver
+        if obj is a Solver object, return a copy of the Solver
         if obj is a list, and each element of list is solver resolvable, return list of solvers
         '''
         if isinstance(obj, str):
             return SolverFactory(obj.lower())
         elif callable(getattr(obj, "solve", None)):
-            return obj
+            return deepcopy(obj)
```

Comment:
I am a little nervous about trusting deepcopy and solver objects.  The Pyomo side should be OK, but if the solver has python bindings, then who knows what might happen.  What is the motivation for making a copy of the solver?

Reply:
To preserve the original subsolver time limit setting prescribed by the user (since this setting may now be changed by PyROS in advance of solving a subproblem).

Reply:
I see.  I think it might be safer to create a new solver using the factory.  Maybe something like:
```python
new_solver = SolverFactory(obj.name)
new_solver.options.set_value(obj.options)
return new_solver
```

Reply:
Understood. How should I handle solvers which do not have a `name` attribute (such as GAMS solver types)?

Reply:
Eww.  You are right.  This is a mess: the 'name' is not necessarily correlated with the registration in the SolverFactory, and the old servers are not at all consistent with how they store / manage ephemeral options.  Maybe the better answer is to just use the solver the user gave us and remember / restore the timeout on exit (either normal or error / exception)?

Reply:
See subsequent commits